libs/SmartSync/SmartSync/Classes/Manager/SFSmartSyncSyncManager.m

```diff
@@ -710,23 +580,23 @@ - (void)resumeSyncUpOneEntry:(SFSyncState*)sync
     
     switch(action) {
         case SFSyncUpTargetActionCreate:
-            [target createOnServer:objectType fields:fields completionBlock:completeBlockCreate failBlock:failBlock];
+            [target createOnServer:self record:record fieldlist:sync.options.fieldlist completionBlock:completeBlockCreate failBlock:failBlock];
             break;
         case SFSyncUpTargetActionUpdate:
-            [target updateOnServer:objectType objectId:objectId fields:fields completionBlock:completeBlockUpdate failBlock:failBlockUpdate];
+            [target updateOnServer:self record:record fieldlist:sync.options.fieldlist completionBlock:completeBlockUpdate failBlock:failBlockUpdate];
             break;
         case SFSyncUpTargetActionDelete:
             // if locally created it can't exist on the server - we don't need to actually do the deleteOnServer call
-            if ([record[kSyncManagerLocallyCreated] boolValue]) {
+            if ([target isLocallyCreated:record]) {
                 completeBlockDelete(record);
             }
             else {
-                [target deleteOnServer:objectType objectId:objectId completionBlock:completeBlockDelete failBlock:failBlockDelete];
+                [target deleteOnServer:self record:record completionBlock:completeBlockDelete failBlock:failBlockDelete];
             }
             break;
         default:
             // Action is unsupported here.  Move on.
-            [self log:SFLogLevelInfo format:@"%@ unsupported action with value %d.  Moving to the next record.", NSStringFromSelector(_cmd), action];
+            LogSyncInfo(@"%@ unsupported action with value %d.  Moving to the next record.", NSStringFromSelector(_cmd), action);
```

Comment:
Per warnings, `%d` should be `%lu`, and `action` should be cast as `(unsigned long)`.